pilot/pkg/networking/core/accesslog.go

```diff
@@ -236,13 +236,13 @@ func buildAccessLogFilter(f ...*accesslog.AccessLogFilter) *accesslog.AccessLogF
 	}
 }
 
-func (b *AccessLogBuilder) buildListenerFileAccessLog(mesh *meshconfig.MeshConfig) *accesslog.AccessLog {
+func (b *AccessLogBuilder) buildListenerFileAccessLog(mesh *meshconfig.MeshConfig, proxyVersion *model.IstioVersion) *accesslog.AccessLog {
 	if cal := b.cachedListenerFileAccessLog(); cal != nil {
```

Comment:
do we need to handle the caching aspect here? maybe only use/set the cache if we are on GE123 so we don't need to handle two caches

Reply:
Ahh, it's a shared access log builder! Good catch; agreed - we should only set the cache on one version